bin/varnishd/cache/cache_backend.c

```diff
@@ -235,7 +235,9 @@ vbe_dir_finish(VRT_CTX, VCL_BACKEND d)
 	if (PFD_State(pfd) != PFD_STATE_USED)
 		assert(bo->htc->doclose == SC_TX_PIPE ||
 		    bo->htc->doclose == SC_RX_TIMEOUT);
-	if (bo->htc->doclose != SC_NULL || bp->proxy_header != 0) {
+	if (bo->htc->doclose != SC_NULL || bp->proxy_header != 0 ||
+	    http_GetHdrField(bo->bereq, H_Connection, "close", NULL) ||
+	    http_GetHdrField(bo->beresp, H_Connection, "close", NULL)) {
```

Comment:
good catch, but shouldn't we better make sure to set `bo->htc->doclose` appropriately before?

Reply:
done!